geowave-store/src/main/java/mil/nga/giat/geowave/store/filter/SpatialQueryFilter.java

```diff
@@ -188,7 +193,7 @@ public void fromBinary(
 					fieldId));
 		}
 		buf.get(theRest);
-		queryGeometry = GeometryUtils.geometryFromBinary(geometryBinary);
+		queryGeometry = geometryInterner.intern(GeometryUtils.geometryFromBinary(geometryBinary));
```

Comment:
fancy! But in looking at JTS geometry I'm not convinced that it will work - I saw this in the source:
"#equals(Object) and #hashCode are not overridden, so that when two topologically equal Geometries are added to HashMaps and HashSets, they remain distinct. This behaviour is desired in many cases."